File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/consent/InMemoryUserConsentGateway.java
package com.cleverferret.v2.feature.ai.consent;

import com.cleverferret.v2.feature.ai.model.AiRequest;

public final class InMemoryUserConsentGateway implements UserConsentGateway {
    private final boolean consentGranted;

    public InMemoryUserConsentGateway(boolean consentGranted) {
        this.consentGranted = consentGranted;
    }

    @Override
    public boolean hasOutboundConsent(AiRequest request) {
        return consentGranted;
    }
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/consent/InMemoryUserConsentGateway.kt
package com.cleverferret.v2.feature.ai.consent

import com.cleverferret.v2.feature.ai.model.AiRequest

class InMemoryUserConsentGateway(private val consentGranted: Boolean) : UserConsentGateway {
    override fun hasOutboundConsent(request: AiRequest): Boolean = consentGranted
}
